### PR TITLE
fix the way the celery task retries

### PR DESCRIPTION
**Pull request by brad**
@orcasgit/orcas-developers I think I managed to get the task retry working. Looks like I misunderstood the method signature when I wrote that and the unit tests weren't testing that the retry actually succeeds.
I have now modified the test so that it verifies the task gets successfully retried.

### CODE REVIEW

File: fitapp/tests/test_retrieval.py [hunk 1]
```diff
@@ -179,20 +179,37 @@ def test_subscription_update_locked(self, mock_add, get_fitbit_data):
...
+            # Delete the cache lock after the first try and adjust the
+            # get_fitbit_data mock to be successful
+            cache.delete(lock_id)
+            get_fitbit_data.side_effect = None
+            get_fitbit_data.return_value = [{
+                'dateTime': self.date,
+                'value': '34'
+            }]
+            raise exc
+        get_fitbit_data.side_effect = side_effect
```
Comment (percyperez) [inline fitapp/tests/test_retrieval.py]: @brad Let me see if I understand the code. It seems that we are setting `get_fitbit_data.side_effect` to a function but inside the same function we set the same attribute `side_effect` to `None` ? Is that how it works?

Comment (brad) [inline fitapp/tests/test_retrieval.py]: @percyperez Exactly, that way the `side_effect` does not get executed the second time the mock is called.

Comment (grokcode) [inline fitapp/tests/test_retrieval.py]: :+1: Thanks for explaining that.

**Comment by grokcode** [comment]
LGTM :shipit:

**Comment by percyperez** [comment]
LGTM too.

**Pull request merged**